modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/FileEntryPicker.js

```diff
@@ -22,20 +22,39 @@ import React, {useEffect, useState} from 'react';
 
 const FileNamePicker = ({
 	maxFileSize: initialMaxFileSize,
+	maxMimeTypeSize,
 	namespace,
 	validExtensions,
 }) => {
-	const maxFileSize = Number(initialMaxFileSize);
+	let maxFileSize = Number(initialMaxFileSize);
 	const inputId = `${namespace}file`;
 	const [inputValue, setInputValue] = useState('');
 	const [fileName, setFileName] = useState('');
 	const [maxFileSizeError, setMaxFileSizeError] = useState(false);
+	const [maxSize, setMaxSizeValue] = useState(maxFileSize);
 
 	useEffect(() => {
 		setFileName(inputValue ? inputValue.replace(/^.*[\\]/, '') : '');
 	}, [inputValue]);
 
 	const onInputChange = ({target}) => {
+		const jsonObj = JSON.parse(maxMimeTypeSize);
```

Comment:
If it is necessary parse the JSON (I hope not) better outside of the input change function, we don't need to reparse it in each change.

Reply:
based on your idea I updated source to remove redundant code lines. The code is cleaner right now, thank you

Reply:
the new code does not parse the maxMimeTypeSize object to the JSON object right now.